test/suites/edgexfoundry/proxy_test.go

```diff
@@ -0,0 +1,100 @@
+package test
+
+import (
+	"edgex-snap-testing/test/utils"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	snapData        = "/var/snap/edgexfoundry/current/temp"
+	kongAminJwtFile = "/var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt"
+)
+
+func TestAddProxyUser(t *testing.T) {
+	const (
+		publicKey  = snapData + "/public.pem"
+		privateKey = snapData + "/private.pem"
+	)
+
+	// start clean
+	utils.Exec(t, `sudo rm -rf `+snapData)
+
+	t.Cleanup(func() {
+		utils.Exec(t, `sudo rm -rf `+snapData)
+	})
+
+	// Due to confinement issues when running this test, we write the files to SNAP_DATA
+	utils.Exec(t, `sudo mkdir -p `+snapData)
+
+	// Read the API Gateway token
+	kongAdminJWT, _ := utils.Exec(t, "sudo cat "+kongAminJwtFile)
+	require.NotEmpty(t, kongAdminJWT)
+
+	// Create private and public keys
+	utils.Exec(t, `sudo openssl ecparam -genkey -name prime256v1 -noout -out `+privateKey)
+	utils.Exec(t, `sudo openssl ec -in `+privateKey+` -pubout -out `+publicKey)
+
+	// Use secrets-config to add a user example with id 1000
+	// On success, the above command prints the user id
+	stdout, _ := utils.Exec(t, `
+			sudo edgexfoundry.secrets-config proxy adduser \
+			--token-type jwt --user example --algorithm ES256 \
+			--public_key `+publicKey+` --id 1000 -jwt `+kongAdminJWT)
+	require.Equal(t, "1000\n", stdout)
```

Comment:
The test is incomplete. It is adding a user but not using it to query any endpoint.